### PR TITLE
Added docstring coding convention and specified line terminator to developer docs

### PR DESCRIPTION
**Pull request by innovate-invent**
I have created this to queue a conversation about expanding on coding conventions for this project.
I fully expect this to change before being pulled.

### CODE REVIEW

File: docs/source/developer/contributing.rst [hunk 1]
```diff
@@ -204,6 +204,11 @@ well-defined coding style (would it be nice to follow :pep:`7`?).
 Code and documentation should generally fit within 80 columns, for
 maximum readability with all existing tools (such as code review UIs).
 
+In addition:
+
+* Docstrings should adhere to the [NumPy docstring convention](http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html).
+* Lines are terminated with \r\n (Windows style).
```
Comment (seibert) [inline docs/source/developer/contributing.rst]: There is no need to standardize line terminations in the source code.  Git automatically translates between line endings when checking out on different platforms as long as you use the form that is standard on your platform.

**Comment by innovate-invent** [comment]
Ah, I wasn't aware of this functionality but if I was I would have forced it to LF. This would have created havoc in the commits.

I added a flag in .gitattributes to override any user config.

**Comment by stuartarchibald** [comment]
Thanks for suggesting work on code base clean up. Following on, https://github.com/numba/numba/pull/3495 is for `flake8` compliance. 

WRT this PR, the change cannot be accepted at present as it mutates the file permission mode. The change to `.gitattributes` more generally needs discussion by the core devs as it is potentially dangerous. Similarly the adoption of a docstring style requires discussion. Will raise this at the next core developer meeting. Thanks again.

**Comment by stuartarchibald** [comment]
Thanks for writing this patch for Numba. This PR is now unfortunately stale with respect to the current master branch. If you wish to resubmit for review please do so by reopening this PR and rebasing the patch(es) against current master. Thanks for your help in improving Numba!

**Pull request closed without merging**